docs/writing-sql.md

```diff
@@ -0,0 +1,156 @@
+---
+title: SQL best practices
+parent: Library
+nav_order: 5
+# audience: clinical researcher or engineer familiar with project
+# type: reference
+---
+
+This doc contains low-level advice on SQL patterns that will help
+avoid errors and make queries more performant.
+
+# General Syntax
+
+Our primary target is
+[Amazon Athena](https://aws.amazon.com/athena/),
+which is based off of
+[Trino](https://trino.io/);
+Searching for topics on either of these DBs is a good way to debug syntax issues
+not mentioned the scope of this document. The 
+[Trino functions reference](https://trino.io/docs/current/functions.html)
+is particularly useful.
+
+We use
+[DuckDB](https://duckdb.org/)
+for our unit tests, and it could be used as a datastore locally if desired. If there
+is a difference between DuckDb and Athena/Trino, we patch DuckDB with a
+[user defined function](https://duckdb.org/docs/api/python/function.html).
+You can see some examples of this in the
+[DuckDB DatabaseBackend](https://github.com/smart-on-fhir/cumulus-library/blob/main/cumulus_library/databases/duckdb.py#L37).
+Hopefully you will not have to do this. If you do, we are probably interested
+in supporting it, so please reach out and let us know.
+
+## Dates
+
+FHIR dates/times come across the wires as ISO 8061-formatted strings.
+To get them into actual timestamps, use the `from_is08061_timestamp` function as
+part of your query, and case to either `date` or `timestamp`.If you want to get a
+specific portion of a date, like a month or year, use the `date_trunc` function. 
+An example use case:
+
+```sql
+ SELECT 
+        date(from_iso8601_timestamp(dr.date)) AS doc_date,
+        date_trunc('day', date(from_iso8601_timestamp(dr."context"."period"."start")))
+            AS author_day,
```

Comment:
That might be an odd example, right - does `date_trunc('day', date_value)` actually do anything? Maybe truncate to month?